From 305cd5794a59ffc5e52c89fb3714530a142cc084 Mon Sep 17 00:00:00 2001
From: Simon Pintarelli <simonpi@daint-ln004.cscs.ch>
Date: Wed, 14 Jan 2026 09:33:14 +0100
Subject: [PATCH] fix compilation flags

---
 flags.guess | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/flags.guess b/flags.guess
index 32aa9dc..5a676d8 100644
--- a/flags.guess
+++ b/flags.guess
@@ -116,7 +116,7 @@ case $my_FC in
 # with some versions of gfortran/gcc you make get errors about unused functions when compiling with the options below;
 # if so, either change -Wunused to -Wunused -Werror=no-unused-function, or remove -Wunused, or remove -Werror
 #
-        DEF_FFLAGS="-std=gnu -fimplicit-none -fmax-errors=10 -pedantic -pedantic-errors -Waliasing -Wampersand -Wcharacter-truncation -Wline-truncation -Wsurprising -Wno-tabs -Wunderflow -ffpe-trap=invalid,zero,overflow -Wunused" # -mcmodel=medium
+        DEF_FFLAGS="-std=gnu -fimplicit-none -fmax-errors=10 -fallow-argument-mismatch -Waliasing -Wampersand -Wcharacter-truncation -Wline-truncation -Wsurprising -Wno-tabs -Wunderflow -ffpe-trap=invalid,zero,overflow -Wunused" # -mcmodel=medium
         OPT_FFLAGS="-O3 -finline-functions"
         DEBUG_FFLAGS="-g -O0 -ggdb -fbacktrace -fbounds-check -frange-check -Werror"
         # useful to track loss of accuracy because of automatic double to single precision conversion:  -Wconversion  (this may generate many warnings...)
-- 
2.35.3

