--- a/xt_base/qtinterf/qtinterf.cc
+++ b/xt_base/qtinterf/qtinterf.cc
@@ -38,6 +38,7 @@
  $Id:$
  *========================================================================*/
 
+#include "config.h"
 #include "qtinterf.h"
 #include "qtfile.h"
 #include "qtfont.h"
@@ -1721,6 +1722,20 @@ QTbag::SetErrorLogName(const char *fname)
 }
 
 
+#ifndef HAVE_MOZY
+
+// Resolve help pop-up when mozy is not included.
+//
+bool
+QTbag::PopUpHelp(const char*)
+{
+    PopUpErr(MODE_ON, "Help system is not available in this executable.");
+    return (false);
+}
+
+#endif
+
+
 // Static function.
 // Return a QColor, initialized with the given attr color.  Used
 // with the QTbag popups, main color functions are in QTdev.
