--- a/cpp/basix/mdspan.hpp
+++ b/cpp/basix/mdspan.hpp
@@ -2791,7 +2791,7 @@ struct layout_stride {
 
       MDSPAN_TEMPLATE_REQUIRES(
         class IntegralType,
-        (std::is_convertible<IntegralType, typename extents_type::index_type>::value)
+        (MDSPAN_IMPL_TRAIT(std::is_convertible, IntegralType, typename extents_type::index_type))
       )
       MDSPAN_INLINE_FUNCTION
       // Need to avoid zero length c-array

