diff --git a/src/sys/objects/device/impls/segmentedmempool.hpp b/src/sys/objects/device/impls/segmentedmempool.hpp
index df7cfc48914..624fe22c39b 100644
--- a/src/sys/objects/device/impls/segmentedmempool.hpp
+++ b/src/sys/objects/device/impls/segmentedmempool.hpp
@@ -128,7 +128,7 @@ private:
   bool            open_      = true;                              // is this chunk open?
   int             stream_id_ = device::DefaultStream::INVALID_ID; // id of the last stream to use the chunk, populated on release
   size_type       size_      = 0;                                 // size of the chunk
-  const size_type start_     = 0;                                 // offset from the start of the owning block
+  size_type       start_     = 0;                                 // offset from the start of the owning block
 
   template <typename U>
   PETSC_NODISCARD bool stream_compat_(const device::StreamBase<U> *) const noexcept;
